docs: Add a "Skip navigation" link on the /help and /api route

madrix01 opened the pull request:
PR for the issue #15948 

**I have added the following changes.**
- When pressed tab the button appears from the top
- And when we press enter, it takes us to top of the page.
- I have remove the hash in URL after we press enter.

Please review the commit and suggest the changes. This is my first issue.

![image](https://user-images.githubusercontent.com/51414879/129312684-1259ca76-7203-4b32-bd57-c48e8852898c.png)

madrix01 commented:
@timabbott @zulipbot can you please review this #19571 and suggest some changes. This is my first pr.

abhijeetbodas2001 commented on templates/zerver/portico-header.html:
```diff
@@ -1,8 +1,19 @@
             {% if custom_logo_url %}
-            <a class="brand logo" href="{{ root_domain_uri }}/"><img draggable="false" src="{{ custom_logo_url }}" class="portico-logo" alt="{{ _('Zulip') }}" content="Zulip" /></a>
+            <a class="brand logo" href="{{ root_domain_uri }}/"><img draggable="false" src="{{ custom_logo_url }}" class="portico-logo" alt="{{ _('Zulip') }}"  content="Zulip" /></a>
             {% else %}
```
Why this extra space here?

madrix01 replied:
I did it by mistake. I removed it.

abhijeetbodas2001 commented on static/js/portico/help.js:
```diff
@@ -10,6 +10,10 @@ function registerCodeSection($codeSection) {
+    // Activating skip-link
+    const $skiplink = $codeSection.find(".skip-link");
+    $skiplink.addClass("active");
+
```
The comment here seems unnecessary, since it's clear from reading the code what it does (which is a nice thing!)

abhijeetbodas2001 commented on docs/git/using.md:
```diff
@@ -48,6 +48,7 @@ from Zulip's main repositories.
 it. Rather than using `git pull`, which by default is a shortcut for `git fetch
 && git merge FETCH_HEAD` ([docs][gitbook-git-pull]), you should use `git fetch`
+
 and then `git rebase`.
```
Unrelated change?

madrix01 replied:
Yes, it was unrelated change. It won't happen again. I have reverted it back.

abhijeetbodas2001 commented on static/styles/portico/portico.css:
```diff
@@ -702,6 +701,31 @@ input.text-error {
+    }
+}
+
 .app {
```
Can you talk about how you have tested this CSS (and made sure to avoid duplication) in the commit message?

madrix01 replied:
I have read about the documentation of HTML and CSS. But since I have only added one class which is unique. But i have checked which fields are affecting the issue. If there is any can you suggest me ?

abhijeetbodas2001 replied:
What I meant was, you should check if there are any existing classes which you can reuse. And defining totally new colors colors should be done thoughtfully, so as to avoid duplication. I don't have any specific suggestions right now.

madrix01 replied:
Ok, I'll check for the classes which are similar to the one I need.  In the button I have added some animation due to which I created new class. But for my future issue I'll make sure that there are no duplicate css used.

abhijeetbodas2001 reviewed (commented):
@madrix01 Thanks for working on this! I left some comments on cleanliness and Git discipline. Please read https://zulip.readthedocs.io/en/latest/contributing/version-control.html#commit-messages to know more about writing commit messages and using Git. You should also probably squash your commits into a single commit.

abhijeetbodas2001 commented:
@madrix01 You should squash your commits into a single commit. Check out https://zulip.readthedocs.io/en/latest/git/fixing-commits.html#squashing-commits (and the related sections).

madrix01 commented:
@abhijeetbodas2001 Hello, I have squashed the commit as per the docs and squashed 9 commits into one. After I pushed the code. It is not passing a CI test about codeql. I have changed nothing in the code you can check changes. What is causing this error?

abhijeetbodas2001 commented:
I don't have an idea what this error is caused by. Please wait for a bit, someone will definitely help you out on chat.zulip.org

madrix01 commented:
Ok, there is no change in the code. It's the same as the previous commits.

aryanshridhar reviewed (commented):
Looks like you haven't properly configured your git and probably that's what causing the error. 
[This](https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration) might help.

madrix01 commented:
I have edited git config and it passed all the CI tests. Please have a look at it @abhijeetbodas2001 .

madrix01 commented:
I have made the changes with my git config. Now all the CI test are passing. Thank you @aryanshridhar @abhijeetbodas2001 .

madrix01 commented:
Due to my internal error in my git I'm closing this pull request, I'll be creating new one. Sorry for inconvenience.
